examples/common/pigweed/device_service/device_service.proto

```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+
+package chip.rpc;
+
+import 'proto/empty.proto';
+
+// type lengths defined in chip spec 8.2.3.1
+message DeviceInfo {
+  uint32 vendor_id = 1;
+  uint32 product_id = 2;
+  uint32 software_version = 3;
+  optional string serial_number = 4;
```

Comment:
Do we need to use the `optional` keyword, which is a newish proto3 feature?  If no code will ever call a `has_serial_number` method, it doesn't give us anything as far as I know.

Reply:
I'm still trying to understand what we gain by marking `serial_number` optional.  This does not change the RPC interface.  Callers may include serial_number or not.  If they do not, `serial_number` will hold the implicit proto3 string default, which is an empty string.

What this does is that it allows consuming code to discern whether `serial_number` was in fact set or not.  Normally, you can only discern this with non-primitive proto3 types.  If it is important for consuming code to know this, then optional makes sense.  But what is the intent then for consuming code?  Will it throw an error?  Will it use some other CHIP-specific default serial number?

Reply:
I removed it for now, and we will see how the tests evolve. My thought was that the serial_number might not always be available right now and having a simple way for the tests to check and optionally populate seemed cleaner then checking against the empty string.

Reply:
OK, thank you for the explanation.